core-primitives/types/src/light_client_init_params.rs

```diff
@@ -0,0 +1,41 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use codec::{Decode, Encode};
+use sp_finality_grandpa::AuthorityList;
+use sp_std::vec::Vec;
+
+// The variants will be chosen according to availability of grandpa authorities on the parent chain.
+#[derive(Encode, Decode)]
+pub enum LightClientInitParams<Header> {
+	Grandpa { genesis_header: Header, authorities: AuthorityList, authority_proof: Vec<Vec<u8>> },
+}
+
+impl<Header> LightClientInitParams<Header> {
+	pub fn get_authorities(&self) -> &AuthorityList {
+		match self {
+			LightClientInitParams::Grandpa { authorities, .. } => authorities,
+		}
+	}
```

Comment:
Shouldn't we in this case return an empty list?

Reply:
My personal preference would be an option, but I think both are fine. You can leave it like that.

Reply:
Since we're returning a reference to a list, an option would probably be better, no? Because otherwise we need to store an empty list somewhere in order to return that reference.

Reply:
At the moment we don't even need this method  since the respective logic changed. When in case we will need it, we will use Option.